components/camel-huaweicloud-functiongraph/src/main/java/org/apache/camel/FunctionGraphProducer.java

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    private void invokeFunction(Exchange exchange) {
+
+        // convert exchange body to Map object
+        Object body = exchange.getMessage().getBody();
+        Map request;
+        if (body == null) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Exchange body is null");
+            }
+            throw new IllegalArgumentException("Exchange body is mandatory and should be a valid Map or JSON string");
+        } else if (body instanceof Map) {
+            request = exchange.getMessage().getBody(Map.class);
+        } else {
+            String strBody = exchange.getMessage().getBody(String.class);
+            try {
+                request = new ObjectMapper().readValue(strBody, HashMap.class);
+            } catch (JsonProcessingException e) {
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Invalid response body given");
+                }
+                throw new IllegalArgumentException("Request body must be a JSON or a HashMap");
+            }
+        }
+
+        // checking for function name and function package
+        if (ObjectHelper.isEmpty(clientConfigurations.getFunctionName())) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error("Function name not found.");
+            }
+            throw new IllegalArgumentException("Function name is mandatory for invokeFunction.");
+        }
+        if (ObjectHelper.isEmpty(exchange.getProperty(FunctionGraphProperties.FUNCTION_PACKAGE))
+                && ObjectHelper.isEmpty(endpoint.getFunctionPackage())) {
+            if (LOG.isWarnEnabled()) {
+                LOG.warn("Function package not found. Continuing to invoke function with 'default' function package");
+            }
+        }
+
+        // invoke the function
+        InvokeFunctionRequest invokeFunctionRequest = new InvokeFunctionRequest()
+                .withBody(request)
+                .withFunctionUrn(FunctionGraphUtils.composeUrn(FunctionGraphConstants.URN_FORMAT, clientConfigurations))
+                .withXCFFRequestVersion(FunctionGraphConstants.REQUEST_VERSION);
+
+        if (ObjectHelper.isNotEmpty(clientConfigurations.getXCffLogType())) {
+            invokeFunctionRequest.withXCffLogType(clientConfigurations.getXCffLogType());
+        }
+
+        InvokeFunctionResponse response = functionGraphClient.invokeFunction(invokeFunctionRequest);
+        String responseBody = FunctionGraphUtils.extractJsonFieldAsString(response.getResult(), FunctionGraphConstants.RESPONSE_BODY);
+        exchange.getMessage().setBody(responseBody);
+        if (ObjectHelper.isNotEmpty(clientConfigurations.getXCffLogType())) {
+            exchange.setProperty(FunctionGraphProperties.XCFFLOGS, response.getLog());
+        }
+        LOG.info("Invoke Function results: " + response);
+    }
+
+    /**
+     * Initialize the client
+     */
+    private void initClient() {
```

Comment:
This should be done in the endpoint. I don't think you'll change the client options many times. So it makes sense to do this just one time in endpoint and use client instance in producer.

Reply:
Done